core/modules/util/threadSafe.h

```diff
@@ -0,0 +1,101 @@
+// -*- LSST-C++ -*-
```

Comment:
atomic doesn't have a condition variable. It might be good for Sequential and Flag, but not for FlagNotify.